core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -1924,8 +1926,10 @@ class ParallelReduce<FunctorType, Kokkos::TeamPolicy<Properties...>,
               .impl_internal_space_instance()
               ->resize_team_scratch_space(
                   static_cast<std::int64_t>(m_scratch_size[1]) *
-                  (static_cast<std::int64_t>(Cuda::concurrency() /
```

Comment:
This expression expands to: `m_scratch_size[1] * (m_maxThreadsPerSM * cudaProp.multiProcessorCount / (m_team_size * m_vector_size))`.

For V100, this is: `m_scratch_size[1] * (2048 * 80 / (m_team_size * m_vector_size))`.

What is `m_scratch_size[1]` - the PerTeam scratch space requested by the user?

Reply:
m_scratch_size[1] is the scratch size per team calculated from PerTeam and PerThread*team_size

Reply:
Shouldn't it then be
```
resize_team_scratch_space(
                  static_cast<std::int64_t>(m_scratch_size[1]) *
                  (std::min(
                      static_cast<std::int64_t>(Cuda::concurrency() /
                                                (m_team_size * m_vector_size)),
                      static_cast<std::int64_t>(m_league_size))));
```
instead of
```
resize_team_scratch_space(
                  static_cast<std::int64_t>(m_scratch_size[1]) *
                  (std::min(
                      static_cast<std::int64_t>(Cuda::concurrency() /
                                                (m_team_size * m_vector_size)),
                      static_cast<std::int64_t>(m_league_size * m_team_size))));
```
?`

Reply:
@masterleinad Yes, that is what I thought and wrote initially in this PR. This caused a crash in CUDA, so I thought that my understanding of `m_scratch_size[1]` must have been wrong. Now I am unsure why that was causing a crash.

Reply:
With this change, I get the following failure:
```
The following tests FAILED:
          6 - KokkosCore_UnitTest_Cuda3 (SEGFAULT)
```

It segfaults on this test `cuda.team_scratch_1_streams`, which according to valgrind is due to this :

```
==134120== Invalid read of size 8
==134120==    at 0x6D8A3B: Kokkos::Impl::HostInaccessibleSharedAllocationRecordCommon<Kokkos::CudaSpace>::get_record(void*) (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x6D8EBD: Kokkos::Impl::SharedAllocationRecordCommon<Kokkos::CudaSpace>::deallocate_tracked(void*) (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x6DE28B: std::_Function_handler<void (Kokkos::Impl::CudaInternal*), Kokkos::Cuda::Cuda(CUstream_st*, bool)::{lambda(Kokkos::Impl::CudaInternal*)#1}>::_M_invoke(std::_Any_data const&, Kokkos::Impl::CudaInternal*&&) (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x414A41: Test::Impl::cuda_stream_scratch_test(int, int, int, Kokkos::View<long, Kokkos::CudaSpace>) (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x41521D: Test::cuda_team_scratch_1_streams_Test::TestBody() (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x6A61D9: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x6960DF: testing::Test::Run() [clone .part.618] (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x696399: testing::TestInfo::Run() [clone .part.619] (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x696B9C: testing::TestSuite::Run() [clone .part.620] (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x69F1A3: testing::internal::UnitTestImpl::RunAllTests() (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x69F5AE: testing::UnitTest::Run() (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==    by 0x407E0F: main (in /home/jliffla/kokkos-build/core/unit_test/KokkosCore_UnitTest_Cuda3)
==134120==  Address 0xf4248 is not stack'd, malloc'd or (recently) free'd
```